src/test/java/study/StringTest.java

```diff
@@ -0,0 +1,50 @@
+package study;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.*;
+
+public class StringTest {
+    // 요구사항 1
+    @Test
+    void split() {
+        String inputTest;
+        String[] result;
+
+        inputTest = "1,2";
+        result = inputTest.split(",");
+        assertThat(result).containsExactly("1", "2");
+
+        inputTest = "1";
+        result = inputTest.split(",");
+        assertThat(result).contains("1");
+    }
+
+    // 요구사항 2
+    @Test
+    void substring() {
+        String inputTest = "(1,2)";
+        String result = inputTest.substring(1, 4);
+        assertThat(result).contains("1,2");
+    }
+
+    // 요구사항 3
+    @Test
+    @DisplayName("StringIndexOutOfBoundsException 테스트")
+    void charAt() {
+        String inputTest = "abc";
+        assertThatThrownBy(() -> {
+            /*
+                질문1
+                이 안의 코드는 무조건 예외처리되나요?
+                inputTest.charAt(1);
+                위 처럼 정상 코드를 넣어도 에러로 처리되는데 정상적인건지 이해가 되지않아 질문드립니다.
+                무조건 예외처리라면 코드 실행중에 문제가있을때 예외가 아닌 일부로 예외를 낸 코드를 테스트 하는것으로 이해됩니다.
+                답변 주시면 감사하겠습니다 :)
+            */
+            inputTest.charAt(3);
+        }).isInstanceOf(IndexOutOfBoundsException.class)
+                .hasMessageContaining("Index:2, Size: 2");
+    }
```

Comment:
아시다시피 assertThatThrownBy() 테스트 메서드는 예외 발생 여부를 검증하고자 사용하는 메서드입니다 :)
말씀하시는 식 안의 코드에서 특정 코드를 실행했을때 예외가 발생할 경우 테스트가 통과가 되는데요.
현재 정상 코드를 넣어도 테스트가 실패된다고 하시기에 다음 에러 로그를 확인하면 좋을 것 같아요!
```
java.lang.AssertionError: 
Expecting throwable message:
  <"String index out of range: 3">
to contain:
  <"Index:2, Size: 2">
but did not.
```
현제 인덱스를 벗어난 예외가 발생했는데 예외 메시지가 "Index:2, Size: 2"라고 포함되길 기대가 되지만
실제로는 "String index out of range: 3" 메시지가 출력되므로 메시지가 일치하지 않아 테스트가 실패하고 있습니다!

다음과 같이 변경하시면 테스트가 통과가 될 것 같아요 :)

```suggestion
    @Test
    @DisplayName("StringIndexOutOfBoundsException 테스트")
    void charAt() {
        String inputTest = "abc";
        assertThatThrownBy(() -> { inputTest.charAt(3);})
                .isInstanceOf(IndexOutOfBoundsException.class)
                .hasMessageContaining("String index out of range: 3");
    }
```

Reply:
넵 내용 확인했습니다. 친절한 설명 감사합니다. 👍